application/database/migrations/2023_06_05_055427_alter_roles_table_add_is_root.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('roles', function (Blueprint $table) {
+            $table->boolean('is_root')->default(false);
+        });
+
+        // Only one role inside institution can be marked as is_root
+        DB::statement(<<<'EOT'
+            CREATE UNIQUE INDEX roles_is_root_unique ON roles (is_root, institution_id)
+            WHERE is_root = true
+        EOT);
```

Comment:
I was aware of this clause, but I decided to not add it, since we want to only have one root role for a institution in every case and it doesn't matter if the is_root is in deleted pool or not deleted pool. Moreover the it shouldn't be able to delete the root role

Reply:
OK, that makes sense, consider it **[non-blocker]**